crates/rune-cli/src/main.rs

```diff
@@ -45,159 +45,192 @@
 //! [Rune Language]: https://github.com/rune-rs/rune
 //! [runestick]: https://github.com/rune-rs/rune
 
-use anyhow::{bail, Result};
+use anyhow::Result;
+use argh::FromArgs;
 use rune::termcolor::{ColorChoice, StandardStream};
 use rune::EmitDiagnostics as _;
-use std::env;
 use std::fs;
 use std::io;
 use std::path::{Path, PathBuf};
 use std::sync::Arc;
 
 use runestick::{Item, Unit, Value, VmExecution};
 
+/// Rune Programming Language.
+/// CLI Arguments
+#[derive(Debug, Clone, FromArgs)]
+struct Args {
+    /// rune Entry File
+    #[argh(positional)]
+    path: PathBuf,
+    /// provide detailed tracing for each instruction executed.
+    #[argh(switch)]
+    trace: bool,
+    /// dump everything.
+    #[argh(switch, short = 'd')]
+    dump: bool,
+    /// dump default information about unit.
+    #[argh(switch)]
+    dump_unit: bool,
+    /// dump unit instructions.
+    #[argh(switch)]
+    dump_instructions: bool,
+    /// dump the state of the stack after completion. If compiled with `--trace` will dump it after each instruction.
+    #[argh(switch)]
+    dump_stack: bool,
+    /// dump dynamic functions.
+    #[argh(switch)]
+    dump_functions: bool,
+    /// dump dynamic types.
+    #[argh(switch)]
+    dump_types: bool,
+    /// dump native functions.
+    #[argh(switch)]
+    dump_native_functions: bool,
+    /// dump native types.
+    #[argh(switch)]
+    dump_native_types: bool,
+    /// include source code references where appropriate (only available if -O debug-info=true).
+    #[argh(switch)]
+    with_source: bool,
+    /// enabled experimental features.
+    #[argh(switch)]
+    experimental: bool,
+    #[argh(subcommand)]
+    compiler_options: CompilerOptions,
```

Comment:
It's like you to figure out a way to stick with using `Options::parse_option`, or similar pattern. The compiler is intended to be integrated at a low level into other projects, and I'd rather only maintain options parsing in one location in case they want to provide a pass through mode.

I realize that this gives better diagnostics (these are the options available). And I'd like improvements here as well.

Reply:
IMO, Keeping `CompilerOptions` as a subcommand offers a lot of advantages:
1. It is easier to pass options without typo (the CLI Parser will refuse it directly)
2. it is easy to add new options .. just one line to the subcommand and the parser handle the rest.
3. It offers a better and organized documentation and a help message

Reply:
Yes, but those benefit can't be shared by other subsystems that will need to parse compile options, these include (but are not limited to):
* Compiler options passed to the language server.
* A future package manager.
* Third party projects embedding the compiler.

So everything we do related to parsing compiler options has to be integrated in these environments, and keeping two things in sync is error prone, so I want to avoid that. Adding the help text in the current cli was already hacky 😅.

I'd suggest you add a generic help message and a separate command which lists compiler options that uses something from `Options` to get help about the list of available options.

Reply:
Well, if we implemented `FromStr` to the `Options` we could use it directly as subcommand without any wrappers. I will do this update now.